README.md

```diff
@@ -452,12 +453,20 @@ By default `KNAPSACK_PRO_REPOSITORY_ADAPTER` variable has no value so knapsack_p
 
 You can also use git as repository adapter to determine branch and commit hash, please see below section.
 
-#### When should you set global variable `KNAPSACK_PRO_REPOSITORY_ADAPTER=git` (required when CI provider is not supported)
+#### When should you set global variable `KNAPSACK_PRO_REPOSITORY_ADAPTER=git` (when CI provider is not supported and you use git)
 
 `KNAPSACK_PRO_REPOSITORY_ADAPTER` - When it has the value `git`, your local version of git on CI server will be used to get the branch name and commit hash. You also need to set `KNAPSACK_PRO_PROJECT_DIR` with the project directory path.
 
 `KNAPSACK_PRO_PROJECT_DIR` - Path to the project on the CI node, for instance `/home/ubuntu/my-app-repository`. It should be the top-level directory of your repository.
 
+#### When you don't use git
+
+If your CI provider does not expose commit hash and branch name as environment variable then `knapsack_pro` gem does not know what is commit hash and branch name.
```

Comment:
typo: it should either be "through an environment variable" (if we would expect one variable for those valyes), or "through environment variables" (if two).

I would change the second part to ", then `knapsack_pro` does not know these values" to avoid repetition.